core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -366,6 +366,16 @@ object SparkSubmit extends CommandLineUtils with Logging {
       localPyFiles = Option(args.pyFiles).map {
         downloadFileList(_, targetDir, sparkConf, hadoopConf, secMgr)
       }.orNull
+
+      if (clusterManager == STANDALONE || clusterManager == LOCAL) {
+        // Use local files for standalone client mode.
+        args.primaryResource = localPrimaryResource
```

Comment:
Isn't this handled by the code in L522 already?

Reply:
It doesn't seem so.

Reply:
Can you explain what breaks without this line?

The code in L522 is adding the local copy of the primary resource to the user's classpath, which should be all that's needed, right?

Reply:
Thanks for your input. It seems the issue is caused by a conflict with our internal patch. I will close this PR for now.